lib/zlib.js

```diff
@@ -728,31 +728,31 @@ Zlib.prototype.params = function params(level, strategy, callback) {
 // generic zlib
 // minimal 2-byte header
 function Deflate(opts) {
-  if (!(this instanceof Deflate))
+  if (!new.target)
```

Comment:
Refs: https://github.com/nodejs/node/issues/42042#issuecomment-1044346436
```suggestion
  if (new.target === undefined)
```